### PR TITLE
Refactor diagnostics for perf

### LINKED ISSUE #115: Flow checks for CSS files?
Hello,

Flow type coverage is checking also for CSS files which is bad. I think you need to include ONLY `.js` files. As you can see in picture I opened `app.css` file and I get `Flow type coverage` and by pressing it I get warnings (green lines below) which says which code is uncovered...

screenshot from 2017-04-22 11-25-02

P.S. Same happens to JSON files... It seems it checks for ALL files, not only JS.

### LINKED ISSUE #116: Flow works even if I don't use flow annotation
Hello,

Flow is working even if I'm don't using `// @flow` annotation... 

screenshot from 2017-04-22 13-25-13

screenshot from 2017-04-22 13-25-32

When I delete that `a;` and save, errors disappears. And this happens for all `.js` documents, by the way the funniest part is, in `[ignore]` is this `__tests__` dir, there's this file `app.test.js`, but I even tested with other `.js` files it's the same... So this plugin somehow ignores even `[ignore]` in `.flowconfig` and you don't need type `// @flow` or `/* @flow */` to enable type check, funny :smile:

### PR DESCRIPTION
**Pull request by thymikee**
**Summary**

* Fixed `hasFlowPragma` check regex
* Run `flow` only if `@flow` pragma is detected
* Debounce all requests to flow with 500ms timeout (matches `eslint` behavior)
* Run prettier through `flowDiagnostics`
* Delete `.flowconfig` from playground, as it unabled flow check (not sure why though)
* Add `yarn.lock`

Overall this should drastically reduce number of requests to flow server when `runOnEdit` is enabled.

Fixes #116 
Fixes #115

cc @orta

### CODE REVIEW

File: lib/flowDiagnostics.js [hunk 1]
```diff
@@ -1,243 +1,250 @@
...
+    })
+  );
+
+  // Update diagnostics when document is edited
+  subscriptions.push(
+    vscode.workspace.onDidChangeTextDocument(event => {
+      const isDocumentActive = activeTextEditor.document === event.document;
+
+      if (isDocumentActive && isRunOnEditEnabled() && hasFlowPragma(event.document.getText())) {
+        debouncedUpdateDiagnostics(context, event.document);
```
Comment (orta) [inline lib/flowDiagnostics.js]: nice touch 👍

File: .vscode/settings.json [hunk 2]
```diff
@@ -2,12 +2,13 @@
 {
 	"files.exclude": {
 		// "build": true, // set this to true to hide the "build" folder with the compiled JS files,
-		"npm-debug.log**": true
+		"npm-debug.log": true
```
Comment (orta) [inline .vscode/settings.json]: This would show the logs I think?

Comment (thymikee) [inline .vscode/settings.json]: Right, changed to single star

File: lib/utils/util.js [hunk 3]
```diff
@@ -33,12 +33,7 @@ export function toURI(path:string):Uri {
 }
 
 export function hasFlowPragma(content:string) {
-	const hasPragma =
-		content.startsWith('/* @flow */') ||
-		content.startsWith('// @flow\n') ||
-		/^\s*\/\*+\s*@flow\s*\*+\//m.test(content) ||
-		/^\s*\/\/\s*@flow\s*$/m.test(content)
-	return hasPragma
+	return /^\s*(\/*\*+|\/\/)\s*@flow/m.test(content);
```
Comment (orta) [inline lib/utils/util.js]: @Gozala do you know if we're going to miss out of things but this switch?

Comment (thymikee) [inline lib/utils/util.js]: Previous regex didn't catch things like
```js
/*
 * @flow
 */
```
which are pretty common.
I tried to cover most cases out there: http://regexr.com/3fumm

We could further optimise checking for pragma on larger files while typing, by caching the result and revalidating if the pragma is still there once in a while.

**Comment by svipben** [comment]
@orta When extension will be updated with this PR?

**Comment by orta** [comment]
OK, I got half an hour - gonna give this a local clone and a run through myself then if it's 👍 I'll merge and prepare for the next release.

**Comment by orta** [comment]
Looking good 👍 

screen shot 2017-05-17 at 21 06 17

This extension is definitely not cleanup up after itself, but I don't think that's the responsibility of this PR, as it's been known for a long time. Let's get this in and published.

**Comment by orta** [comment]
Bah, this repo needs Danger - no CHANGELOG :D

**Comment by thymikee** [comment]
It need quite a lot of goodies, like tests for example XD

**Comment by orta** [comment]
Hah - yeah 📟

**Pull request merged**